src/PowerShellEditorServices/Services/DebugAdapter/Handlers/ConfigurationDoneHandler.cs

```diff
@@ -97,8 +100,26 @@ private async Task LaunchScriptAsync(string scriptToLaunch)
             {
                 ScriptFile untitledScript = _workspaceService.GetFile(scriptToLaunch);
 
-                await _powerShellContextService
-                    .ExecuteScriptStringAsync(untitledScript.Contents, true, true).ConfigureAwait(false);
+                if (BreakpointApiUtils.SupportsBreakpointApis)
+                {
+                    // Parse untitled files with their `Untitled:` URI as the file name which will cache the URI & contents within the PowerShell parser.
+                    // By doing this, we light up the ability to debug Untitled files with breakpoints.
+                    // This is only possible via the direct usage of the breakpoint APIs in PowerShell because
+                    // Set-PSBreakpoint validates that paths are actually on the filesystem.
+                    ScriptBlockAst ast = Parser.ParseInput(untitledScript.Contents, untitledScript.DocumentUri, out Token[] tokens, out ParseError[] errors);
+
+                    // This seems to be the simplest way to invoke a script block (which contains breakpoint information) via the PowerShell API.
+                    var cmd = new PSCommand().AddScript("& $args[0]").AddArgument(ast.GetScriptBlock());
```

Comment:
```suggestion
                    var cmd = new PSCommand().AddScript(". $args[0]").AddArgument(ast.GetScriptBlock());
```
For consistency the scriptblock should be invoked in the current scope.

Reply:
fixed